server/src/main/java/io/druid/client/ImmutableDruidDataSource.java

```diff
@@ -73,14 +69,52 @@ public boolean isEmpty()
     return segmentsHolder;
   }
 
+  public DataSegment getSegment(String segmentIdentifier)
+  {
+    return idToSegments.get(segmentIdentifier);
+  }
+
   @Override
   public String toString()
   {
-    // partitionNames is intentionally ignored because it is usually large
+    // idToSegments is intentionally ignored because it is usually large
     return "ImmutableDruidDataSource{"
            + "name='" + name
            + "', segments='" + segmentsHolder
            + "', properties='" + properties
            + "'}";
   }
+
+  @Override
+  public boolean equals(Object o)
```

Comment:
in a future PR this is changed to just compare the metadata. I was having trouble getting some tests to pass otherwise.

Reply:
nothing need be done here.

Reply:
I guess you may refer to ImmutableDruidServer? ImmutableDruidDataSource doesn't have metadata.

Reply:
Probably, my bad